### PR TITLE
XWIKI-19172: Display rendered images in view mode inside a lightbox

### PR DESCRIPTION
**Pull request by oanalavinia**
* try to fix hoverImage() flickering problems, that might be caused by the fact that after a refresh the JS was not yet fully loaded

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-lightbox/xwiki-platform-image-lightbox-test/xwiki-platform-image-lightbox-test-docker/src/test/it/org/xwiki/image/lightbox/test/ui/LightboxIT.java [hunk 1]
```diff
@@ -80,6 +80,7 @@ void disabledLightbox(TestUtils testUtils, TestReference testReference, TestConf
 
         // Make sure that the images are displayed.
         testUtils.getDriver().navigate().refresh();
+        lightboxPage = new LightboxPage();
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-lightbox/xwiki-platform-image-lightbox-test/xwiki-platform-image-lightbox-test-docker/src/test/it/org/xwiki/image/lightbox/test/ui/LightboxIT.java:83]: There two lines are repeated a lot, imo that's the sign that they should be moved to a method.

Comment (vmassol) [reply xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-lightbox/xwiki-platform-image-lightbox-test/xwiki-platform-image-lightbox-test-docker/src/test/it/org/xwiki/image/lightbox/test/ui/LightboxIT.java:83]: more than method, it should be moved to a PO class.

Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-lightbox/xwiki-platform-image-lightbox-test/xwiki-platform-image-lightbox-test-docker/src/test/it/org/xwiki/image/lightbox/test/ui/LightboxIT.java:83]: Another approach could be to call `waitUntilPageJSIsLoaded` in methods of `LightboxPage` that are actually requiring the javascript to be loaded to work properly.

File: xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-lightbox/xwiki-platform-image-lightbox-test/xwiki-platform-image-lightbox-test-pageobjects/src/main/java/org/xwiki/image/lightbox/test/po/LightboxPage.java [hunk 2]
```diff
@@ -41,6 +41,12 @@
  */
 public class LightboxPage extends ViewPage
 {
+    public void reloadPage()
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-lightbox/xwiki-platform-image-lightbox-test/xwiki-platform-image-lightbox-test-pageobjects/src/main/java/org/xwiki/image/lightbox/test/po/LightboxPage.java:44]: Some documentation of this method and I'm +1 to merge :)

File: xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-lightbox/xwiki-platform-image-lightbox-test/xwiki-platform-image-lightbox-test-pageobjects/src/main/java/org/xwiki/image/lightbox/test/po/LightboxPage.java [hunk 3]
```diff
@@ -41,6 +41,15 @@
  */
 public class LightboxPage extends ViewPage
 {
+    /**
+     * Make sure that the javascript is also loaded after page refresh since lightbox actions depend on it.
+     */
+    public void reloadPage()
+    {
+        getDriver().navigate().refresh();
+        waitUntilPageJSIsLoaded();
```
Comment (mflorea) [inline xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-lightbox/xwiki-platform-image-lightbox-test/xwiki-platform-image-lightbox-test-pageobjects/src/main/java/org/xwiki/image/lightbox/test/po/LightboxPage.java:50]: This method seems generic so why not put it in some base class (like ``BasePage``)?

**Review by mflorea — CHANGES_REQUESTED** [review]

File: xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/BasePage.java [hunk 4]
```diff
@@ -281,6 +281,16 @@ public BasePage waitUntilPageIsLoaded()
         return this;
     }
 
+    /**
+     * Refresh the page and wait for the javascript to be also loaded.
+     */
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/BasePage.java:286]: Missing a `@since`.

**Comment by mflorea** [comment]
@oanalavinia XWIKI-19172 is closed and released so you shouldn't use it anymore for new commits (next time either create a new issue if the changes are significant, or commit without an issue if they are very small with no effect on the end user). I'll change the commit message when merging, this time.

**Review by mflorea — APPROVED** [review]

**Pull request merged**